docs(rapid prototyping): Add guidance for rapid prototyping (#4697)
patternfly/patternfly-org: packages/documentation-site/patternfly-docs/content/PatternFly-AI/rapid-prototyping/new-prototypes.md

---
id: Rapid prototyping
section: PatternFly-AI
source: new-prototypes
sortValue: 2
---

# Starting new prototypes

> This guide covers setting up new PatternFly prototypes using AI-assisted development tools.

Reviewer: ```suggestion
This guide provides instructions for setting up new PatternFly prototypes using AI-assisted development tools.
```

> 

Reviewer: ```suggestion

Before you get started, review [best practices for AI-assisted prototyping](/patternfly-ai/rapid-prototyping/rapid-prototyping#best-practices-for-ai-assisted-prototyping).

If you need to add AI tools to existing codebases instead, follow our guidelines for [enhancing existing projects](/patternfly-ai/rapid-prototyping/enhancing-existing-projects).
```

Moving the "next steps" section up here

## Project setup

> For new prototypes, we recommend using the **patternfly-react-seed** application as your starting point. This seed application comes pre-configured with:

Reviewer: ```suggestion
For new prototypes, we recommend using the [patternfly-react-seed](https://github.com/patternfly/patternfly-react-seed) application as your starting point. This application provides AI tools with immediate access to PatternFly context and guidelines, which offers better support for vibe coding.

The patternfly-react-seed comes pre-configured with:
```

> - A basic PatternFly React application structure
> - The **patternfly-ai-coding tool** already integrated and ready to use
> - Essential dependencies and build configuration
> - Example components and patterns to reference

Reviewer: ```suggestion
- A basic PatternFly React application structure.
- The patternfly-ai-coding tool already integrated and ready to use.
- Essential dependencies and build configuration.
- Example components and patterns to reference.
```

> The patternfly-react-seed provides an optimal foundation for [vibe coding](/patternfly-ai/rapid-prototyping/rapid-prototyping#what-is-vibe-coding) because the AI tools can immediately access PatternFly context and guidance through the embedded patternfly-ai-coding tool.

Reviewer: ```suggestion
```

> ## Getting the seed application

Reviewer: ```suggestion
### Set up the patternfly-react-seed
```

> 1. Clone or download the patternfly-react-seed repository
> 2. Install dependencies according to the project's README
> 3. Start the development server
> 4. Begin prototyping with your AI development tools

Reviewer: ```suggestion
1. Clone or download the patternfly-react-seed repository.
2. Install dependencies according to [the project's README](https://github.com/patternfly/patternfly-react-seed?tab=readme-ov-file#patternfly-seed).
3. Start the development server.
4. Begin prototyping with your AI development tools.
```

> ## Workflow steps

Reviewer: ```suggestion
## How to prototype 
```

> 1. **Set up your project**: Start with the PatternFly seed application that includes the AI coding tool
> 2. **Define your prototype scope**: Clearly articulate what you want to build using [vibe coding principles](/patternfly-ai/rapid-prototyping/rapid-prototyping#effective-prompting-with-vibe-coding) - describe the experience, not the technical implementation
> 3. **Leverage AI assistance**: Use AI tools to generate PatternFly components based on your requirements, communicating through natural language descriptions of user needs and design intent
> 4. **Iterate and refine**: Review generated code and make adjustments as needed, continuing to use vibe coding to guide refinements
> 5. **Test and validate**: Ensure your prototype meets design and functional requirements

Reviewer: ```suggestion
Once you've installed and set up the patternfly-react-seed, with the patternfly-ai-coding tool, follow these steps to being prototyping: 

1. **Define your scope**: Clearly articulate what you want to build using [vibe coding principles](/patternfly-ai/rapid-prototyping/rapid-prototyping#effective-prompting-with-vibe-coding). Describe the experience, not the technical implementation.
2. **Leverage AI assistance**: Use AI tools to generate PatternFly components based on your requirements, communicating through natural language descriptions of user needs and design intent.
3. **Iterate and refine**: Review generated code and make adjustments as needed, continuing to use vibe coding to guide refinements.
4. **Test and validate**: Ensure your prototype meets design and functional requirements.
```

> ## Next steps
> 
> - Review [best practices for AI-assisted prototyping](/patternfly-ai/rapid-prototyping/rapid-prototyping#best-practices-for-ai-assisted-prototyping)
> - Learn about [enhancing existing projects](/patternfly-ai/rapid-prototyping/enhancing-existing-projects) if you need to add AI tools to current codebases

Reviewer: ```suggestion
```
